### DISCUSSION TITLE
Azure Storage Account with ADLS Gen2 enable

### DISCUSSION THREAD

**Discussion by abb84** [pulumi/pulumi-azure-native#2053] [Q&A]
How to create an Azure Storage Account with ADLS Gen2 enable?

**Comment by zeenmc** [comment]
Same as regular storage account just you need to add one option **isHnsEnabled: true**,

Hsn == hierarchical namespace

This is for example :

```    const storageAccount = new storage.StorageAccount("storageaccountname", {
        accountName: "storageaccountname",
        resourceGroupName: resourceGroup.name,
        sku: {
            name: storage.SkuName.Standard_LRS,
        },
        kind: storage.Kind.StorageV2,
        allowBlobPublicAccess: false,
        allowSharedKeyAccess: true,
        keyPolicy: {
            keyExpirationPeriodInDays: 180,
        },
        minimumTlsVersion: "TLS1_2",
        enableHttpsTrafficOnly: true,
        isHnsEnabled: true,

    });```

**Comment by YuriiVarvynets** [comment]
How do I modify ACLs, owner, group for the created container with pulumi-azure-native?
That’s how to do it via Azure portal https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-acl-azure-portal

Could someone show me the way how do it with pulumi?
